components/places/src/ffi.rs

```diff
@@ -43,6 +48,155 @@ impl UniffiCustomTypeWrapper for Url {
     }
 }
 
+impl UniffiCustomTypeWrapper for Guid {
+    type Wrapped = String;
+
+    fn wrap(val: Self::Wrapped) -> uniffi::Result<Guid> {
+        Ok(Guid::new(val.as_str()))
+    }
+
+    fn unwrap(obj: Self) -> Self::Wrapped {
+        obj.into()
+    }
+}
+
+impl UniffiCustomTypeWrapper for Timestamp {
+    type Wrapped = i64;
+
+    fn wrap(val: Self::Wrapped) -> uniffi::Result<Timestamp> {
+        Ok(Timestamp::from(val as u64))
+    }
+
+    fn unwrap(obj: Self) -> Self::Wrapped {
+        obj.as_millis() as i64
+    }
+}
+
+///*
+/// A bookmark node.
+///
+/// We use a single message type for bookmarks. It covers insertion, deletion,
+/// and update, and represents all three bookmark types.
+///
+/// This simplifies the FFI by reducing the number of types that must go across
+/// it, and retuces boilderplate, but removes some static-ish guarantees we
+/// might have otherwise.
+///
+/// Note that these docs comments are internal, and don't necessarily impact the actual
+/// API we expose to Kotlin/Swift (this is particularly true around reads).
+#[derive(Clone, PartialEq)]
+pub struct BookmarkNode {
+    ///*
+    /// The type of this bookmark, a `BookmarkType` (from `types.rs`).
+    ///
+    /// This impacts which fields may be present.
+    ///
+    /// It's illegal to attempt to change this when updating a bookmark.
+    ///
+    /// Note: this probably should be an `enum`, but prost seems to get upset
```

Comment:
This comment is now stale - and it would indeed seem ideal if we can get an enum in place for this. The existing `BookmarkType` would be ideal, but the fact it's declared with a `val` may or may not make this tricky. Either way, the comment should change in some way :)